### ISSUE TITLE
Formatter for number input type

### ISSUE THREAD

**Issue by bjola** [angular/angular.js#12737]
The function numberInputType in angular.js only checks if the input value is typeof 'number', which lead to strange errors in our application, like:
"Expected `11` to be a number"
Changing it from 
if (!isNumber(value)) {
        throw ngModelMinErr('numfmt', 'Expected `{0}` to be a number', value);
      }

to 

if (!isNumber(value) && isNaN(value)) {
        throw ngModelMinErr('numfmt', 'Expected `{0}` to be a number', value);
      }
solved the problem.

It should work, since it's doing value = value.toString() at the end anyway

**Comment by Narretz** [comment]
What type os your model? It should be a Number.

**Comment by bjola** [comment]
The problem occurred when changing the input type of the hour and minute field of the timepicker to number instead of text. Maybe it has something to do with the model of those fields. 
However, it would make sense to have those fields as number inputs. What else could you put there? :-) 
It also brings up the correct keyboard on phones and tablets.

**Comment by Narretz** [comment]
The docs say that a model for input number must be of type Number. Otherwise the error is thrown. That means you must transform the value yourself. Or do you mean something else?

**Comment by bjola** [comment]
I noticed that. So the question ended up being: Why isn't the input type set to "number" for the timepicker input fields in ui-angular-tpls? They are set to "text", which makes no sense. When trying to change that, this problem occured and was solved by the change mentioned above. So I guess that the timepickers internal hours and minutes models aren't Number types (which is kind of strange).
Maybe that's an angular-ui question that doesn't belong here.
